### ISSUE TITLE
Segfault in tutorial on turn 5 or so

### ISSUE THREAD

**Issue by mesilliac** [wesnoth/wesnoth#2258]
So previously the tutorial was crashing consistently on turn 4 at `assert(false && "Found null handler in handler list");` in game_events/manager.cpp but with this removed it's now crashing occasionally on turn 5-6 with a segmentation fault.

backtrace:

```
#0  0x0000000000cf5d32 in std::_Rb_tree<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::pair<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const, config_attribute_value>, std::_Select1st<std::pair<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const, config_attribute_value> >, std::less<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::allocator<std::pair<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const, config_attribute_value> > >::find(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) const ()
#1  0x0000000000ceec00 in config::operator[](std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) const ()
#2  0x00000000008a5af9 in game_events::event_handlers::remove_event_handler(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) ()
#3  0x00000000005ec495 in int dispatch<&game_lua_kernel::intf_remove_event>(lua_State*) ()
#4  0x0000000000cc9104 in luaD_precall(lua_State*, lua_TValue*, int) ()
#5  0x0000000000ce11ed in luaV_execute(lua_State*) ()
#6  0x0000000000cc9387 in luaD_callnoyield(lua_State*, lua_TValue*, int) ()
#7  0x0000000000cca2b0 in luaD_pcall(lua_State*, void (*)(lua_State*, void*), void*, long, long) ()
#8  0x0000000000cc0469 in lua_pcallk(lua_State*, int, int, int, long, int (*)(lua_State*, int, long)) ()
#9  0x00000000005f96c4 in luaW_pcall(lua_State*, int, int, bool) ()
#10 0x00000000005e47e8 in game_lua_kernel::run_wml_action(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, vconfig const&, game_events::queued_event const&) ()
#11 0x0000000000bd4ed4 in game_events::event_handler::handle_event(game_events::queued_event const&, game_lua_kernel&) ()
#12 0x00000000008abe4d in game_events::wml_event_pump::process_event(std::shared_ptr<game_events::event_handler>&, game_events::queued_event const&) ()
#13 0x00000000008ac136 in std::_Function_handler<void (game_events::manager&, std::shared_ptr<game_events::event_handler>&), game_events::wml_event_pump::operator()()::{lambda(game_events::manager&, std::shared_ptr<game_events::event_handler>&)#1}>::_M_invoke(std::_Any_data const&, game_events::manager&, std::shared_ptr<game_events::event_handler>&) ()
#14 0x00000000008a321f in game_events::manager::execute_on_events(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::function<void (game_events::manager&, std::shared_ptr<game_events::event_handler>&)>) ()
#15 0x00000000008af979 in game_events::wml_event_pump::operator()() ()
#16 0x00000000008affdf in game_events::wml_event_pump::fire(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, game_events::entity_location const&, game_events::entity_location const&, config const&) ()
#17 0x0000000000a98b5e in play_controller::finish_turn() ()
#18 0x0000000000a9d8ad in play_controller::play_turn() ()
#19 0x0000000000aa50d1 in playsingle_controller::play_scenario_main_loop() ()
#20 0x0000000000aa6610 in playsingle_controller::play_scenario(config const&)
    ()
#21 0x00000000008bfafb in campaign_controller::playsingle_scenario(end_level_data&) ()
#22 0x00000000008bff2b in campaign_controller::play_game() ()
#23 0x000000000049e394 in game_launcher::launch_game(game_launcher::RELOAD_GAME_DATA) ()
#24 0x000000000042db94 in main ()
```

**Comment by Vultraz** [comment]
Should be fixed by 238a994429c2.

**Comment by mesilliac** [comment]
looks good to me
